src/main/java/net/neoforged/neoforge/transfer/handlers/templates/InfiniteResourceHandler.java

```diff
@@ -0,0 +1,81 @@
+/*
+ * Copyright (c) NeoForged and contributors
+ * SPDX-License-Identifier: LGPL-2.1-only
+ */
+
+package net.neoforged.neoforge.transfer.handlers.templates;
+
+import java.util.Objects;
+import net.neoforged.neoforge.transfer.handlers.resources.ISingleResourceHandler;
+import net.neoforged.neoforge.transfer.resources.IResource;
+import net.neoforged.neoforge.transfer.transaction.TransactionContext;
+
+/**
+ * An {@link ISingleResourceHandler} that allows extraction of an unlimited amount of a specified resource.
+ *
+ * @param <T> The type of resource that this storage can accept.
+ */
+public class InfiniteResourceHandler<T extends IResource> implements ISingleResourceHandler<T> {
+    /**
+     * Resource that should be provided infinitely. Mustn't be {@code null}.
+     */
+    public T infinite;
+
+    public InfiniteResourceHandler(T resource) {
+        this.infinite = resource;
+    }
+
+    @Override
+    public int insert(T resource, int amount, TransactionContext context) {
+        return 0; // doesn't allow insertions
+    }
+
+    @Override
+    public int extract(T resource, int amount, TransactionContext context) {
+        return resource.equals(infinite) ? amount : 0;
+    }
+
+    @Override
+    public T getResource(int index) {
+        Objects.checkIndex(index, size());
```

Comment:
The index checking is very erratic in this class for whether you actually bother validating the index or not...